File: DataLayer/Models/Bug.cs
using DataLayer.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace DataLayer.Models
{
    public class Bug
    {
        public int Id { get; set; }

        public string Name { get; set; }
        public string Description { get; set; }
        public Urgency Priority { get; set; }
        public Criticality Severity { get; set; }
        public Status Status { get; set; }

        public User Creator { get; set; }
        public DateTime DateCreate { get; set; }

        public ICollection<History> Histories { get; set; }

        public Bug() { }

        public Bug(string name, User creator, string description = "Некоторое описание ошибки", 
            Urgency priority = Urgency.Fourth, Criticality severity = Criticality.Low, Status status = Status.New)
        {
            Name = name;
            Description = description;
            Priority = priority;
            Severity = severity;
            Status = status;

            Creator = creator;
            DateCreate = DateTime.Now;
        }
    }
}

File: Bugs.Frontend/GUI/bugs/logic/models/Bug.ts
import {Urgency} from "./Urgency";
import {Criticality} from "./Criticality";
import {Status} from "./Status";
import {User} from "../../../users/logic/models/User";

export class Bug {
    id: number;
    name: string;
    priority: Urgency;
    reproSteps: string;
    severity: Criticality;
    status: Status;
    statusComment: string;
    creator = new User(null);
    dateCreate: string;
    histories = [];
    
    constructor(bug: Bug) {
        if (bug == null) {
            this.id = 0;
            this.name = "";
            this.priority = Urgency.Fourth;
            this.reproSteps = "";
            this.severity = Criticality.Low;
            this.status = Status.New;
            this.statusComment = "";
            this.creator = new User(null);
            this.dateCreate = "";
            this.histories = [];
        }
        else {
            this.id = bug.id;
            this.name = bug.name;
            this.priority = bug.priority;
            this.reproSteps = bug.reproSteps;
            this.severity = bug.severity;
            this.status = bug.status;
            this.statusComment = bug.statusComment;
            this.creator = bug.creator;
            this.dateCreate = bug.dateCreate;
            this.histories = bug.histories;
        }
    }
}